File: LLM-Translations/GPT-4/codenet/Java/Python/s898738430.py
import sys

def main():
    n = int(input().strip())
    a = list(map(int, input().strip().split()))
    ans = 0
    for i in range(1, n-1):
        if a[i] != min(a[i-1], min(a[i], a[i+1])) and a[i] != max(a[i-1], max(a[i], a[i+1])):
            ans += 1
    print(ans)

if __name__ == "__main__":
    main()

File: LLM-Translations/GPT-4/codenet/Java/Go/s898738430.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n int
	fmt.Scan(&n)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}
	ans := 0
	for i := 1; i < n-1; i++ {
		if a[i] != min(a[i-1], min(a[i], a[i+1])) && a[i] != max(a[i-1], max(a[i], a[i+1])) {
			ans++
		}
	}
	fmt.Println(ans)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// 
